neo/Core/TX/Transaction.py

```diff
@@ -712,7 +712,10 @@ def GetScriptHashesForVerifying(self):
                 if type(attr.Data) is UInt160:
                     hashes.add(attr.Data)
                 else:
-                    hashes.add(UInt160(data=attr.Data))
+                    if len(attr.Data) == 40:
```

Comment:
previously this branch expected a `bytes` object. Imagine we add a `--tx-attr` of type `script` (0x20), then we would have to add something like: 
> --tx-attr={"usage":32,"data": b'#\xba\'\x03\xc52c\xe8\xd6\xe5"\xdc2 39\xdc\xd8\xee\xe9'}

this will fail to parse correctly because there's a space in the above data value, even though it translates to a valid script_hash. This little branch here allows to provide a data input on the console as follows:
> --tx-attr={"usage":32,"data": b'23ba2703c53263e8d6e522dc32203339dcd8eee9'}

Reply:
I looked into this suggestion. Assuming you were referring to this part
https://github.com/CityOfZion/neo-python/blob/f570ce99302033b6276e338ee5043449b118ba70/neo/Prompt/Utils.py#L133-L139

or this part a little bit deeper
https://github.com/CityOfZion/neo-python/blob/f570ce99302033b6276e338ee5043449b118ba70/neo/Prompt/Utils.py#L162-L168
then we would pretty much be duplicating this loop
https://github.com/CityOfZion/neo-python/blob/f570ce99302033b6276e338ee5043449b118ba70/neo/Core/TX/Transaction.py#L710-L715

as we would still have to determine that the attribute usage is of type `Script` and then that it's not already in the right format. I think the current solution makes most sense, but I'm still open for an example what you might have in mind if you think otherwise.

Reply:
I was thinking of something like this:
```    
        datum = obj['data']
        if type(datum) is str:
            datum = datum.encode('utf-8')
        if len(datum) == 40:
            datum = binascii.unhexlify(datum)
        usage = obj['usage']
        return TransactionAttribute(usage=usage, data=datum)
```

Reply:
Aha. I understand your view now. I do still think that would be suboptimal because any TX attribute of length 40 will be unhexlified. This means also `Description`, `Remark` types and others that happen to be exactly 40 characters. The current suggested solution is 100% just for script hashes, and that's all I (we?) can foresee it applies for now.

Reply:
How about something like this: 
```        datum = obj['data']
        if type(datum) is str:
            datum = datum.encode('utf-8')
        usage = obj['usage']
        if usage == script and len(datum) == 40:
            datum = binascii.unhexlify(datum)

        return TransactionAttribute(usage=usage, data=datum)```